### PR TITLE
dev-python/{ipython,pylint}, net-libs/grpc, dev-cpp/abseil-cpp: add ~riscv keyword

### PR DESCRIPTION
**Pull request by caoye275**
Also add ~riscv keyword to related packages

### CODE REVIEW

**Comment by mkszuba** [comment]
Strong NAK from me until at least the following two issues have been addressed:
 * masking USE=ssl on dev-python/pycurl forces this flag to off in net-misc/curl itself. This not only is IMHO a bad idea but also causes a tug of war over this USE flag when I try to emerge dev-python/ipython with tests enabled for all the packages you have modified;
 * masking tests is STRONGLY discouraged, and if it really cannot be avoided you should provide in your commit messages details of how exactly you have tested the relevant ebuilds.

**Comment by caoye275** [comment]
> Strong NAK from me until at least the following two issues have been addressed:
> 
> * masking USE=ssl on dev-python/pycurl forces this flag to off in net-misc/curl itself. This not only is IMHO a bad idea but also causes a tug of war over this USE flag when I try to emerge dev-python/ipython with tests enabled for all the packages you have modified;
> * masking tests is STRONGLY discouraged, and if it really cannot be avoided you should provide in your commit messages details of how exactly you have tested the relevant ebuilds.

Thanks for your comments!! I'll try to work on these two issues, finding if I can unmask these to USE flags.

**Comment by caoye275** [comment]
> Strong NAK from me until at least the following two issues have been addressed:
> 
> * masking USE=ssl on dev-python/pycurl forces this flag to off in net-misc/curl itself. This not only is IMHO a bad idea but also causes a tug of war over this USE flag when I try to emerge dev-python/ipython with tests enabled for all the packages you have modified;
> * masking tests is STRONGLY discouraged, and if it really cannot be avoided you should provide in your commit messages details of how exactly you have tested the relevant ebuilds.

There're still issues remained for USE=ssl on dev-python/pycurl. I've dropped the commits about this package this time.
Umask USE=test on dev-python/ipython as well as its dependent packages

**Comment by caoye275** [comment]
> Strong NAK from me until at least the following two issues have been addressed:
> 
> * masking USE=ssl on dev-python/pycurl forces this flag to off in net-misc/curl itself. This not only is IMHO a bad idea but also causes a tug of war over this USE flag when I try to emerge dev-python/ipython with tests enabled for all the packages you have modified;
> * masking tests is STRONGLY discouraged, and if it really cannot be avoided you should provide in your commit messages details of how exactly you have tested the relevant ebuilds.

Fix these 2 issues. Thanks for you suggestions :)

File: profiles/arch/riscv/package.use.mask [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Copyright 2019-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
+# Ye Cao <ye.c@rioslab.org> (2021-06-30)
+# # Dependencies not keyworded, not tested
+sci-libs/fftw mpi
+media-sound/alsa-utils libsamplerate
+dev-python/ipython doc matplotlib nbconvert notebook
```
Comment (mkszuba) [inline profiles/arch/riscv/package.use.mask:8]: Why do you mask _doc_ and _matplotlib_? Your PR takes care of all the missing keywords on the dependencies pulled in by these two.

File: profiles/arch/riscv/package.use.mask [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+dev-python/matplotlib cairo excel tk wxwidgets
```
Comment (mkszuba) [inline profiles/arch/riscv/package.use.mask:9]: Don't mask _tk_, the necessary dependencies got keyworded yesterday.

File: profiles/arch/riscv/package.use.mask [hunk 3]
```diff
@@ -1,6 +1,14 @@
 # Copyright 2019-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
+# Ye Cao <ye.c@rioslab.org> (2021-06-30)
+# # Dependencies not keyworded, not tested
+sci-libs/fftw mpi
+media-sound/alsa-utils libsamplerate
+dev-python/ipython doc matplotlib nbconvert notebook
+dev-python/matplotlib cairo excel tk wxwidgets 
+app-text/texlive-core tk xetex
```
Comment (mkszuba) [inline profiles/arch/riscv/package.use.mask:10]: Same here.

File: profiles/arch/riscv/package.use.mask [hunk 4]
```diff
@@ -200,7 +208,7 @@ net-irc/irssi socks5
 net-libs/gnutls doc test-full
 net-libs/ldns doc python
 net-libs/libpcap rdma
-net-misc/curl brotli curl_ssl_mbedtls curl_ssl_nss rtmp ssh nss mbedtls
+net-misc/curl brotli curl_ssl_mbedtls rtmp ssh nss mbedtls
```
Comment (mkszuba) [inline profiles/arch/riscv/package.use.mask:211]: Please remove the _nss_ mask as well.

**Review by mkszuba — CHANGES_REQUESTED** [review]
Much better, thank you! Just a few more mask tweaks and I think this will be ready for merging.

File: profiles/arch/riscv/package.use.mask (hunk 4 above)
Comment (mkszuba) [reply profiles/arch/riscv/package.use.mask:211]: ...which in light of other changes here just committed to the tree means you should simply remove this line.

Comment (caoye275) [reply profiles/arch/riscv/package.use.mask:211]: Actually, I'm not sure if it's correct to remove all this USE flags. Simply remove this line might cause some issues in other places?

File: profiles/arch/riscv/package.use.mask (hunk 2 above)
Comment (caoye275) [reply profiles/arch/riscv/package.use.mask:9]: I think I could unmask this in another commit after this PR?  It's quite hard for me to update my local branch to the latest changes. Solving this seperately after merging will be much easier.

File: profiles/arch/riscv/package.use.mask (hunk 1 above)
Comment (caoye275) [reply profiles/arch/riscv/package.use.mask]: Thank you! I've fixed the issue just now.

Comment (caoye275) [reply profiles/arch/riscv/package.use.mask]: Thank you! fixed it.

File: profiles/arch/riscv/package.use.mask (hunk 2 above)
Comment (caoye275) [reply profiles/arch/riscv/package.use.mask]: Thank you! fixed it.

File: profiles/arch/riscv/package.use.mask (hunk 3 above)
Comment (caoye275) [reply profiles/arch/riscv/package.use.mask]: Thank you! fixed it.

File: profiles/arch/riscv/package.use.mask (hunk 4 above)
Comment (caoye275) [reply profiles/arch/riscv/package.use.mask]: Thank you! fixed it.

**Comment by mkszuba** [comment]
Looking good (cosmetics aside, I just had to propagate the keywords on dev-python/astroid and dev-python/pylint to the new versions which made their way to the tree between your PR submission and now), thank you very much for this huge batch of keywords!

**Comment by caoye275** [comment]
> Looking good (cosmetics aside, I just had to propagate the keywords on dev-python/astroid and dev-python/pylint to the new versions which made their way to the tree between your PR submission and now), thank you very much for this huge batch of keywords!

You're welcome. Also very glad to make some contributions :)

**Pull request closed without merging**